packages/widgetbook/lib/src/knobs/text_knob.dart

```diff
@@ -0,0 +1,118 @@
+import 'package:flutter/material.dart';
+import 'package:provider/provider.dart';
+
+import 'package:widgetbook/src/knobs/knobs.dart';
+import 'package:widgetbook/src/knobs/nullable_checkbox.dart';
+
+class TextKnob extends Knob<String> {
+  TextKnob({
+    required String label,
+    String? description,
+    required String value,
+  }) : super(
+          label: label,
+          description: description,
+          value: value,
+        );
+
+  @override
+  Widget build() => TextKnobWidget(
+        label: label,
+        description: description,
+        value: value,
+      );
+}
+
+class NullableTextKnob extends Knob<String?> {
+  NullableTextKnob({
+    required String label,
+    String? description,
+    required String? value,
+  }) : super(
+          label: label,
+          description: description,
+          value: value,
+        );
+
+  @override
+  Widget build() => TextKnobWidget(
+        label: label,
+        description: description,
+        value: value,
+        nullable: true,
+      );
+}
+
+class TextKnobWidget extends StatefulWidget {
+  const TextKnobWidget({
+    Key? key,
+    required this.label,
+    required this.description,
+    required this.value,
+    this.nullable = false,
+  }) : super(key: key);
+
+  final String label;
+  final String? description;
+  final String? value;
+  final bool nullable;
+
+  @override
+  State<TextKnobWidget> createState() => _TextKnobWidgetState();
+}
+
+class _TextKnobWidgetState extends State<TextKnobWidget> {
+  String value = '';
+  final controller = TextEditingController();
+
+  @override
+  void initState() {
+    super.initState();
+    if (widget.value != null) {
+      controller.text = widget.value!;
```

Comment:
The TextField for the knob looks different than the TextField from the Search (in the `NavigationPanel`). This is likely because the TextField within the search is locally themed. It would be great if you could define the Theme globally to make the used TextFields match the Style 👍🏻